File: standard-udfs-examples/standard-udfs-example-udfs-presto/src/test/java/com/linkedin/stdudfs/examples/presto/TestArrayElementAtFunctionWrapper.java
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.stdudfs.examples.presto;

import com.facebook.presto.operator.scalar.AbstractTestFunctions;
import com.facebook.presto.spi.type.ArrayType;
import com.google.common.collect.ImmutableList;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import static com.facebook.presto.spi.type.BooleanType.*;
import static com.facebook.presto.spi.type.IntegerType.*;
import static com.facebook.presto.spi.type.VarcharType.*;
import static com.facebook.presto.type.UnknownType.*;


public class TestArrayElementAtFunctionWrapper extends AbstractTestFunctions {

  @BeforeClass
  public void registerFunction() {
    registerScalarFunction(new ArrayElementAtFunctionWrapper());
  }

  @Test
  public void testArrayElementAt() {

    assertFunction("cast(array_element_at(array['1', '2'], 1) as varchar)", VARCHAR, "2");

    assertFunction("array_element_at(array[1, 2], 1)", INTEGER, 2);

    assertFunction("array_element_at(array[true, false], 1)", BOOLEAN, false);

    assertFunction("cast(array_element_at(array[array['1'], array['2']], 1) as array(varchar))", new ArrayType(VARCHAR),
        ImmutableList.of("2"));

    assertFunction("array_element_at(null, 1)", UNKNOWN, null);

    assertFunction("array_element_at(array[1], null)", INTEGER, null);
  }
}

File: transportable-udfs-examples/transportable-udfs-example-udfs-spark/src/test/scala/com/linkedin/transport/examples/spark/TestArrayElementAtFunctionWrapper.scala
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.transport.examples.spark

import com.linkedin.transport.spark.common.AssertSparkExpression._
import org.testng.annotations.{BeforeClass, Test}

import scala.collection.mutable.WrappedArray._


class TestArrayElementAtFunctionWrapper {

  @BeforeClass
  def registerFunction(): Unit = {
    registerStandardUdf(
      "array_element_at",
      classOf[ArrayElementAtFunctionWrapper]
    )
  }

  @Test
  def testArrayElementAt(): Unit = {
    assertFunction("array_element_at(array('1', '2'), 1)", "2")
    assertFunction("array_element_at(array(1, 2), 1)", 2)
    assertFunction("array_element_at(array(true, false), 1)", false)
    assertFunction("array_element_at(array(array('1'), array('2')), 1)", make(Array("2")))
    assertFunction("array_element_at(null, 1)", null)
    assertFunction("array_element_at(array(1), null)", null)
  }
}
